pkg/k8shandler/elasticsearch_helper.go

```diff
@@ -101,20 +101,24 @@ func updateAllIndexReplicas(clusterName, namespace string, client client.Client,
 
 	// get list of indices and call updateIndexReplicas for each one
 	for index, health := range indexHealth {
-		// only update replicas for indices that don't have same replica count
-		if numberOfReplicas := parseString("settings.index.number_of_replicas", health.(map[string]interface{})); numberOfReplicas != "" {
-			currentReplicas, err := strconv.ParseInt(numberOfReplicas, 10, 32)
-			if err != nil {
-				return false, err
-			}
+		if _, ok := health.(map[string]interface{}); ok {
+			// only update replicas for indices that don't have same replica count
+			if numberOfReplicas := parseString("settings.index.number_of_replicas", health.(map[string]interface{})); numberOfReplicas != "" {
+				currentReplicas, err := strconv.ParseInt(numberOfReplicas, 10, 32)
+				if err != nil {
+					return false, err
+				}
 
-			if int32(currentReplicas) != replicaCount {
-				// best effort initially?
-				logrus.Debugf("Updating %v from %d replicas to %d", index, currentReplicas, replicaCount)
-				if ack, err := updateIndexReplicas(clusterName, namespace, client, index, replicaCount); err != nil {
-					return ack, err
+				if int32(currentReplicas) != replicaCount {
+					// best effort initially?
+					logrus.Debugf("Updating %v from %d replicas to %d", index, currentReplicas, replicaCount)
+					if ack, err := updateIndexReplicas(clusterName, namespace, client, index, replicaCount); err != nil {
+						return ack, err
+					}
 				}
 			}
+		} else {
+			logrus.Debugf("Improper count for index: \"%s\", health: \"%v\"", index, health)
```

Comment:
Do we want a Debugf here? Or Errorf/Warnf?

Reply:
once elasticsearch startup is complete, this block should not be entered. so it is temporary.

Reply:
If this implies an error on elasticsearch startup, I would lift the severity to error. However, if it is an message accompanying pod *un*-readiness then an info would be more practical.

Reply:
This implies the response wasn't a map which is different then we expect and a condition we believe we should never get to.  I think maybe it's more appropriate to "Warn" and say what the problem was:  

`logrs.Warnf("Unable to evaluate the number of replicas for index %q: %v", index, health")`

We should also mention the cluster name as the operator could be reconciling any elasticsearch cluster, not just the one for logging

Reply:
updated